src/utils.js

```diff
@@ -415,6 +415,18 @@ export const getNanoContractFeatureToggle = (state) => (
  */
 export const getTimestampFormat = (timestamp) => moment.unix(timestamp).format(t`DD MMM YYYY [•] HH:mm`)
 
+/**
+ * Extract the result of a generator function when it is done.
+ */
+export const consumeGenerator = (generator) => {
```

Comment:
This is not "consuming" a generator, it's waiting until it's done and returning the last yielded value

Name is misleading

Reply:
No, it is not returning the last yielded value, it is returning the return value of a generator after it has yield for the last time.

I believe it is "consuming" the generator because it is running the generator until its end.